lib/fastlane/plugin/wpmreleasetoolkit/models/firebase_test_runner.rb

```diff
@@ -0,0 +1,115 @@
+require 'json'
+require 'uri'
+require 'fileutils'
+require 'google/cloud/storage'
+
+module Fastlane
+  class FirebaseTestRunner
+    VALID_TEST_TYPES = %w[instrumentation robo].freeze
+
+    def initialize(key_file:, verify_gcloud_binary: true)
+      raise "Unable to find key file: #{key_file}" unless File.file? key_file
+
+      @key_file = key_file
+      @has_authenticated = false
+
+      FirebaseTestRunner.verify_has_gcloud_binary if verify_gcloud_binary
+    end
+
+    def authenticate_if_needed
+      return if @has_authenticated
+
+      Action.sh(
+        'gcloud', 'auth', 'activate-service-account',
+        '--key-file', @key_file
+      )
+
+      @has_authenticated = true
+    end
+
+    # Run a given APK and Test Bundle on the given device type.
+    #
+    # @param [String] apk_path Path to the application APK on disk.
+    # @param [String] test_apk_path Path to the test runner APK on disk.
+    # @param [FirebaseDevice] device The virtual device to run tests on.
+    # @param [String] type The type of test to run.
+    #
+    def run_tests(apk_path:, test_apk_path:, device:, type: 'instrumentation')
+      raise "Unable to find apk: #{apk_path}" unless File.file? apk_path
+      raise "Unable to find apk: #{test_apk_path}" unless File.file? test_apk_path
+      raise "Invalid Type: #{type}" unless VALID_TEST_TYPES.include? type
```

Comment:
nit: Maybe personal preference, but I find using parentheses for params is better for readability here
```suggestion
      raise "Unable to find apk: #{apk_path}" unless File.file?(apk_path)
      raise "Unable to find apk: #{test_apk_path}" unless File.file?(test_apk_path)
      raise "Invalid Type: #{type}" unless VALID_TEST_TYPES.include?(type)
```